layers/core_checks/cc_render_pass.cpp

```diff
@@ -5104,3 +5113,229 @@ bool CoreChecks::ValidateInheritanceInfoFramebuffer(VkCommandBuffer primaryBuffe
+            if (pLocationInfo->pColorAttachmentLocations[i] != VK_ATTACHMENT_UNUSED &&
+                pLocationInfo->pColorAttachmentLocations[i] >= phys_dev_props.limits.maxColorAttachments) {
+                const Location loc =
+                    loc_info.dot(Struct::VkRenderingAttachmentLocationInfoKHR, Field::pColorAttachmentLocations, i);
+                skip |= LogError("VUID-VkRenderingAttachmentLocationInfoKHR-pColorAttachmentLocations-09515", objlist, loc,
+                                 "pColorAttachmentLocations[%" PRIu32 "] == %" PRIu32
+                                 " is greater than the maxColorAttachments limit(% " PRIu32 ")",
+                                 i, pLocationInfo->pColorAttachmentLocations[i], phys_dev_props.limits.maxColorAttachments);
+            }
+        }
+
+        std::map<uint32_t, uint32_t> unique;
+
+        for (uint32_t i = 0; i < pLocationInfo->colorAttachmentCount; ++i) {
+            const uint32_t location = pLocationInfo->pColorAttachmentLocations[i];
+            if (location == VK_ATTACHMENT_UNUSED) continue;
+
+            if (unique.find(location) != unique.end()) {
+                const Location loc =
+                    loc_info.dot(Struct::VkRenderingAttachmentLocationInfoKHR, Field::pColorAttachmentLocations, i);
+                skip |= LogError("VUID-VkRenderingAttachmentLocationInfoKHR-pColorAttachmentLocations-09513", objlist, loc,
+                                 "pColorAttachmentLocations[%" PRIu32 "] is same as in pColorAttachmentLocations[%" PRIu32
+                                 "] (%" PRIu32 ")",
+                                 i, unique[location], location);
+            } else
+                unique[location] = i;
+        }
+    }
+
+    if (pLocationInfo->colorAttachmentCount > phys_dev_props.limits.maxColorAttachments) {
+        skip |= LogError("VUID-VkRenderingAttachmentLocationInfoKHR-colorAttachmentCount-09514", objlist,
+                         loc_info.dot(Field::colorAttachmentCount),
+                         "(%" PRIu32 ") is greater than the maxColorAttachments limit (%" PRIu32 ").",
+                         pLocationInfo->colorAttachmentCount, phys_dev_props.limits.maxColorAttachments);
+    }
+
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateCmdSetRenderingAttachmentLocationsKHR(VkCommandBuffer commandBuffer,
+                                                                      const VkRenderingAttachmentLocationInfoKHR *pLocationInfo,
+                                                                      const ErrorObject &error_obj) const {
+    const auto &cb_state = *GetRead<vvl::CommandBuffer>(commandBuffer);
+    bool skip = false;
+
+    if (!enabled_features.dynamicRenderingLocalRead) {
+        const LogObjectList objlist(commandBuffer);
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-dynamicRenderingLocalRead-09509", objlist,
```

Comment:
when only one object
```suggestion
        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-dynamicRenderingLocalRead-09509", commandBuffer,
```

Reply:
Done.